File: output/deepseek_chat/avatar/direct_translation/python/java/codeforces_131_B.java
import java.util.*;
import java.io.*;

public class codeforces_131_B {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String[] input = br.readLine().split(" ");
        Map<Integer, Integer> w = new HashMap<>();
        for (String s : input) {
            int num = Integer.parseInt(s);
            w.put(num, w.getOrDefault(num, 0) + 1);
        }
        int c = 0;
        for (Map.Entry<Integer, Integer> entry : w.entrySet()) {
            int key = entry.getKey();
            int value = entry.getValue();
            if (key == 0) {
                c += value * (value - 1);
            } else {
                if (w.containsKey(-key)) {
                    c += value * w.get(-key);
                }
            }
        }
        System.out.println(c / 2);
    }
}

File: output/deepseek_r1/avatar/direct_translation/java/python/codeforces_131_B.py
n = int(input())
cnt = [0] * 21
for x in map(int, input().split()):
    cnt[x + 10] += 1
res = 0
for i in range(10):
    res += cnt[i] * cnt[20 - i]
res += (cnt[10] * (cnt[10] - 1)) // 2
print(res)
